packages/clay-form/src/LocalizedInput.tsx

```diff
@@ -3,144 +3,170 @@
-		languages,
-		onInputValueChange,
-		onLanguageChange,
-		placeholder = 'Text to translate...',
-		selectedLanguage,
-		url,
-		value,
-	}) => (
-		<ClayForm.Group>
-			<label htmlFor="localizedInput">{label}</label>
-
-			<ClayInput.Group>
-				{url && (
-					<ClayInput.GroupItem prepend shrink>
-						<ClayInput.GroupText>{'/'}</ClayInput.GroupText>
+	(
+		{
+			ariaLabels = {
+				default: 'Default',
+				openLanguageDropdown: 'Open Language Dropdown',
+				translated: 'Translated',
+				untranslated: 'Unstranslated',
+			},
+			helpText,
+			id,
+			label = 'Check for translations',
+			locales,
+			onSelectedLocaleChange,
+			onTranslationsChange,
+			placeholder = 'Text to translate...',
+			prependContent,
+			resultFormatter = val => val,
+			selectedLocale,
+			spritemap,
+			translations,
+			...otherProps
+		},
+		ref
+	) => {
+		const [active, setActive] = React.useState(false);
+
+		const defaultLanguage = locales[0];
+
+		return (
+			<ClayForm.Group>
+				{label && <label htmlFor={id}>{label}</label>}
+
+				{helpText && <ClayForm.Text>{helpText}</ClayForm.Text>}
+
+				<ClayInput.Group>
+					{prependContent && (
+						<ClayInput.GroupItem prepend shrink>
+							<ClayInput.GroupText>
+								{prependContent}
+							</ClayInput.GroupText>
+						</ClayInput.GroupItem>
+					)}
+
+					<ClayInput.GroupItem>
+						<ClayInput
+							{...otherProps}
+							id={id}
+							onChange={e => {
+								onTranslationsChange({
+									...translations,
+									[selectedLocale.label]: e.target.value,
+								});
+							}}
+							placeholder={placeholder}
+							ref={ref}
+							value={translations[selectedLocale.label] || ''}
+						/>
 					</ClayInput.GroupItem>
-				)}
-
-				<ClayInput.GroupItem>
-					<ClayInput
-						id="localizedInput"
-						onChange={event =>
-							onInputValueChange(event.target.value)
-						}
-						placeholder={placeholder}
-						value={value}
-					/>
-				</ClayInput.GroupItem>
-
-				<ClayInput.GroupItem shrink>
-					<DropDownWithState
-						hasRightSymbols
-						language={selectedLanguage}
-					>
-						<ClayDropDown.ItemList>
-							{languages.map(
-								(language: {
-									label: string;
-									symbol: string;
-									value: string;
-								}) => (
-									<ClayDropDown.Item
-										key={language.label}
-										onClick={() =>
-											onLanguageChange(language)
-										}
-									>
-										<ClayIcon
-											className="inline-item inline-item-before"
-											spritemap={spritemap}
-											symbol={language.symbol}
-										/>
-
-										{language.label}
-
-										<ClayLabel
-											className="dropdown-item-indicator-end"
-											displayType={
-												language.label ===
-												defaultLanguage.label
-													? 'info'
-													: language.value
-													? 'success'
-													: 'warning'
+
+					<ClayInput.GroupItem shrink>
+						<ClayDropDown
+							active={active}
```

Comment:
Is this always the superior choice compared to what I did by defining `ClayComponentWithState` above the component?

Reply:
Not necessarily _always_, but I think its best if we avoid over abstraction and wait until it's needed. In this case it wasn't super necessary to abstract since we didn't use it in multiple places and it also didn't necessarily help the readability or complexity of the code.